src/main/java/seedu/homerce/model/appointment/Appointment.java

```diff
@@ -114,11 +119,34 @@ public boolean isSame(UniqueListItem other) {
         if (!(other instanceof Appointment)) {
             return false;
         }
-
         Appointment otherAppointment = (Appointment) other;
-        return client.equals(otherAppointment.getClient())
-                && service.equals(otherAppointment.getService())
-                && appointmentDate.equals(otherAppointment.appointmentDate)
-                && timeOfDay.equals(otherAppointment.timeOfDay);
+        if (appointmentDate.equals(otherAppointment.appointmentDate)) {
+            return isClashing(otherAppointment);
+        } else {
+            return false;
+        }
+    }
+
+    private boolean isClashing(Appointment other) {
+        LocalTime startTimeOfThis = getAppointmentTime().getLocalTime();
+        LocalTime startTimeOfOther = other.getAppointmentTime().getLocalTime();
+        if (startTimeOfThis.equals(startTimeOfOther)) {
+            return true;
+        }
+        Duration durationThis = getService().getDuration();
+        Duration durationOther = other.getService().getDuration();
+        LocalTime endTimeOfThis = startTimeOfThis
+            .plusHours((long) Math.floor(durationThis.value))
+            .plusMinutes((long) ((durationThis.value % 1) * 60));
+        LocalTime endTimeOfOther = startTimeOfOther
+            .plusHours((long) Math.floor(durationOther.value))
+            .plusMinutes((long) ((durationOther.value % 1) * 60));
+        System.out.println("Comparing: " + this + " and other: " + other);
```

Comment:
Should this SOUT be removed from the "production" code? Or maybe we could use logging instead?